src/core_modules/capture-core/reducers/descriptions/searchPage.reducerDescription.js

```diff
@@ -10,39 +17,54 @@ export const searchPageStatus = {
     ERROR: 'ERROR',
 };
 
-export const searchPageDesc = createReducerDescription({
-    [searchPageActionTypes.SEARCH_RESULTS_INITIAL_VIEW]: state => ({
-        ...state,
-        searchStatus: searchPageStatus.INITIAL,
-    }),
-    [searchPageActionTypes.SEARCH_RESULTS_SUCCESS_VIEW]: (state, { payload: { searchResults, searchResultsPaginationInfo } }) => ({
-        ...state,
-        searchStatus: searchPageStatus.SHOW_RESULTS,
-        searchResults,
-        searchResultsPaginationInfo,
-    }),
-    [searchPageActionTypes.SEARCH_RESULTS_LOADING_VIEW]: state => ({
-        ...state,
-        searchStatus: searchPageStatus.LOADING,
-    }),
-    [searchPageActionTypes.SEARCH_RESULTS_EMPTY_VIEW]: state => ({
-        ...state,
-        searchStatus: searchPageStatus.NO_RESULTS,
-    }),
-    [searchPageActionTypes.SEARCH_RESULTS_ERROR_VIEW]: state => ({
-        ...state,
-        searchStatus: searchPageStatus.ERROR,
-    }),
-    [searchPageActionTypes.CURRENT_SEARCH_INFO_SAVE]: (state, { payload: { searchScopeType, searchScopeId, formId, currentSearchTerms } }) => ({
-        ...state,
-        currentSearchInfo: { searchScopeType, searchScopeId, formId, currentSearchTerms },
-    }),
-}, 'searchPage', {
+
+export const searchPage = createReducer({
     searchStatus: searchPageStatus.INITIAL,
     searchResults: [],
     currentSearchInfo: {},
     searchResultsPaginationInfo: {
         nextPageButtonDisabled: false,
         currentPage: 0,
     },
+}, (builder) => {
+    builder
+        .addCase(
+            showInitialViewOnSearchPage,
+            (state) => {
+                state.searchStatus = searchPageStatus.INITIAL;
+            })
+        .addCase(
+            showSuccessResultsViewOnSearchPage,
+            (
+                state,
+                { payload: { searchResults, searchResultsPaginationInfo } },
+            ) => {
+                state.searchStatus = searchPageStatus.SHOW_RESULTS;
+                state.searchResults = searchResults;
+                state.searchResultsPaginationInfo = searchResultsPaginationInfo;
+            })
+        .addCase(
+            showLoadingViewOnSearchPage,
+            (state) => {
+                state.searchStatus = searchPageStatus.LOADING;
+            })
+        .addCase(
+            showEmptyResultsViewOnSearchPage,
+            (state) => {
+                state.searchStatus = searchPageStatus.NO_RESULTS;
+            })
+        .addCase(
+            showErrorViewOnSearchPage,
+            (state) => {
+                state.searchStatus = searchPageStatus.ERROR;
+            })
+        .addCase(
+            saveCurrentSearchInfo,
+            (
+                state,
+                { payload: { searchScopeType, searchScopeId, formId, currentSearchTerms } },
+            ) => {
+                state.currentSearchInfo = { searchScopeType, searchScopeId, formId, currentSearchTerms };
+            });
```

Comment:
So there are couple of improvements here. 

First one is annotation. Now we can annotate the state on the component and have it annotated when we use it later on either in epics or containers.

Second is `Immer` `redux-toolkit` comes with `Immer` as enchantment. That means we will always have immutable state without having to do all the destruction with es6. Meaning the `...state` and so on. 

We can now simply assign variables and `Immer` will do the resty for us under the hood. 🎉